sql/compression_defaults.sql

```diff
@@ -0,0 +1,291 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+
+CREATE OR REPLACE FUNCTION _timescaledb_functions.get_segmentby_defaults(
+    relation regclass
+)
+    RETURNS JSONB LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+    _table_name NAME;
+    _schema_name NAME;
+    _hypertable_row _timescaledb_catalog.hypertable;
+    _segmentby NAME;
+    _cnt int;
+BEGIN
+    SELECT n.nspname, c.relname INTO STRICT _schema_name, _table_name
+    FROM pg_class c
+    INNER JOIN pg_namespace n ON (n.oid = c.relnamespace)
+    WHERE c.oid = relation;
+
+    SELECT * INTO STRICT _hypertable_row FROM _timescaledb_catalog.hypertable h WHERE h.table_name = _table_name AND h.schema_name = schema_name;
+
+    --STEP 1 if column stats exist use unique indexes
+    with index_attr as (
+        SELECT
+        a.attnum, min(a.pos) as pos
+        FROM
+            (select indkey, indnkeyatts from pg_catalog.pg_index where indisunique and indrelid = relation) i
+        INNER JOIN LATERAL
+            (select * from unnest(i.indkey) with ordinality) a(attnum, pos) ON (TRUE)
+        WHERE a.pos <= i.indnkeyatts
+        GROUP BY 1
+    )
+    SELECT
+      a.attname INTO _segmentby
+    FROM
+      index_attr i
+    INNER JOIN
+      pg_attribute a on (a.attnum = i.attnum AND a.attrelid = relation)
+    --right now stats are from the hypertable itself. Use chunks in the future.
+    INNER JOIN pg_statistic s ON (s.staattnum = a.attnum and s.starelid = relation)
+    WHERE
+      a.attname NOT IN (SELECT column_name FROM _timescaledb_catalog.dimension d WHERE d.hypertable_id = _hypertable_row.id)
+      AND s.stadistinct > 1
+    ORDER BY i.pos
+    LIMIT 1;
+
+    IF FOUND THEN
+        return json_build_object('columns', json_build_array(_segmentby), 'confidence', 10);
```

Comment:
How do you pick the confidence levels? 1-10? 1-100? Something else? What does 5 mean?

Reply:
added comments